Source/WebKit/WebProcess/Inspector/wpe/WebInspectorUIWPE.cpp

```diff
@@ -36,22 +37,20 @@ void WebInspectorUI::didEstablishConnection()
 {
     static std::once_flag onceFlag;
     std::call_once(onceFlag, [] {
-        const char* libDir = PKGLIBDIR;
+        const char* dataDir = PKGDATADIR;
+        GUniqueOutPtr<GError> error;
+
 #if ENABLE(DEVELOPER_MODE)
-        // Probably no need for a specific env var here. Assume the inspector resources.so file is
-        // in the same directory as the injected bundle lib, for developer builds.
-        const char* path = g_getenv("WEBKIT_INJECTED_BUNDLE_PATH");
+        const char* path = g_getenv("WEBKIT_INSPECTOR_RESOURCES_PATH");
         if (path && g_file_test(path, G_FILE_TEST_IS_DIR))
-            libDir = path;
+            dataDir = path;
 #endif
-        GUniquePtr<char> bundleFilename(g_build_filename(libDir, "libWPEWebInspectorResources.so", nullptr));
-        GModule* resourcesModule = g_module_open(bundleFilename.get(), G_MODULE_BIND_LAZY);
-        if (!resourcesModule) {
-            WTFLogAlways("Error loading libWPEWebInspectorResources.so: %s", g_module_error());
-            return;
+        GUniquePtr<char> gResourceFilename(g_build_filename(dataDir, "inspector.gresource", nullptr));
+        GRefPtr<GResource> gresource = g_resource_load(gResourceFilename.get(), &error.outPtr());
```

Comment:
Here as well, we need to adopt the pointer on construction:

```suggestion
        GRefPtr<GResource> gresource = adoptGRef(g_resource_load(gResourceFilename.get(), &error.outPtr()));
```